dex/src/main/java/io/crate/breaker/RowAccounting.java

```diff
@@ -37,7 +37,7 @@
      * @throws CircuitBreakingException if too much memory would be consumed
      *         after materializing this row.
      */
-    void accountForAndMaybeBreak(Row row);
+    Row accountForAndMaybeBreak(Row row);
```

Comment:
This is a bit non-intuititve. Do we absolutely need it? I get that it's useful for pipe-ing the Row to further API calls but I'm not sure that benefit outweighs the possible confustion. An interface like this indicates it'll return a new Row - and given the name of the method, I'd expect to maybe return a total of how much memory it accounted for.

Reply:
Was a bit skeptical myself. Will change it back.